### ISSUE TITLE
video url not found

### ISSUE THREAD

**Issue by rbjnazz** [CoolnsX/dra-cla#2]
```
Loading episode 5...
https://dembed1.com/streaming.php?id=MjA0ODYz&x=https://hlsx04.dracache.com/newhls/1afc57ae65d12136311bcd8e56c79ccf/ep.5.v0.1657637944.m3u8&title=Flower+of+Evil&typesub=SUB&sub=&cover=Y292ZXIvZmxvd2VyLW9mLWV2aWwucG5n
base64: invalid input
bad decrypt
D85E750274000000:error:1C80006B:Provider routines:ossl_cipher_generic_block_final:wrong final block length:providers/implementations/ciphers/ciphercommon.c:429:

Video URL not found
```

**Comment by farisamirmudin** [comment]
Hello I managed to solve the problem. In order to get the video link, we just need the ID from the URL, which in your case MjA0ODY. But in the script on line 143, this regex command `sed -nE 's/.*id=(.*)&title.*/\1/p'` will subscript the URL starting from id= until title= which would result in `MjA0ODYz&x=https://hlsx04.dracache.com/newhls/1afc57ae65d12136311bcd8e56c79ccf/ep.5.v0.1657637944.m3u8`. 

I notice that for some videos which only have 1 episode , the URL does not have this `&x=` part in them, and the script does work fine on them. Nevertheless, I did some changes that can work with all videos (hopefully). Below are the changes that I made to `decryption_link()`

    decrypt_link() {
  
	  agent="Mozilla/5.0 (X11; Linux x86_64; rv:99.0) Gecko/20100101 Firefox/100.0"
	  id=$(printf "%s" "$1" | cut -d'=' -f2 | cut -d'&' -f1) # This is the command to subscript the URL starting from the first = sign until the first & sign
  
	  secret_key='3933343232313932343333393532343839373532333432393038353835373532'
	  iv='39323632383539323332343335383235'
	  ajax=$(printf '%s' "$id" | openssl enc -e -aes256 -K "$secret_key" -iv "$iv" -a) # Instead of encrypting the dpage_link, encryption is done on the ID
	  data=$(curl -A "$agent" -s -H "X-Requested-With:XMLHttpRequest" "$base_url/encrypt-ajax.php?id=${ajax}" | sed -e 's/{"data":"//' -e 's/"}/\n/' -e 's/\\//g')
	  printf '%s' "$data" | base64 -d 2>/dev/null | openssl enc -d -aes256 -K "$secret_key" -iv "$iv" 2>/dev/null | sed -e 's/\].*/\]/' -e 's/\\//g' | grep -Eo 'https:\/\/[-a-zA-Z0-9@:%._\+~#=][a-zA-Z0-9][-a-zA-Z0-9@:%_\+.~#?&\/\/=]*'
  
	  # ajax_url="$base_url/encrypt-ajax.php"
	  # ajax=$(printf "%s" "$1" | sed -nE 's/.*id=(.*)&title.*/\1/p' | openssl enc -e -aes256 -K "$secret_key" -iv "$iv" | base64)
	  # data=$(curl -s -H "X-Requested-With:XMLHttpRequest" "$ajax_url" -d "id=$ajax" | sed -e 's/{"data":"//' -e 's/"}/\n/' -e 's/\\//g')
	  # printf '%s' "$data" | base64 -d | openssl enc -d -aes256 -K "$secret_key" -iv "$iv" | sed -e 's/\].*/\]/' -e 's/\\//g' | tr '{|}' '\n' | sed -nE 's/\"file\":"([^"]*)".*label.*P.*/\1/p'
	  
    }

**Comment by CoolnsX** [comment]
Fixed in latest commit.. just update on ur side and check if it works..

**Comment by rbjnazz** [comment]
Just updated to version 2.1.1 thanks
